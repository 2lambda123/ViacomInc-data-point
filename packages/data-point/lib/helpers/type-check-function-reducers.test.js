/* eslint-env jest */

const Promise = require('bluebird')

const {
  isString,
  isNumber,
  isBoolean,
  isFunction,
  isError,
  isArray,
  isObject
} = require('./type-check-function-reducers')

function testTypeChecker (typeCheck, goodValue, badValue, type) {
  test('It should return true if type matches', () => {
    expect(typeCheck(goodValue)).toBe(goodValue)
  })
  test(`It should throw error when not matched`, () => {
    return Promise.try(() => typeCheck(badValue))
      .catch(e => e)
      .then(result => {
        expect(result).toBeInstanceOf(Error)
        expect(result._message).toMatchSnapshot()
      })
  })
}

describe('isString', () => {
  testTypeChecker(isString, 'foo', 1, 'string')
})

describe('isNumber', () => {
  testTypeChecker(isNumber, 1, 'foo', 'number')
})

describe('isBoolean', () => {
  testTypeChecker(isBoolean, true, 'foo', 'boolean')
  testTypeChecker(isBoolean, false, 'foo', 'boolean')
})

describe('isFunction', () => {
  const fn = () => {}
  testTypeChecker(isFunction, fn, 'foo', 'function')
})

describe('isError', () => {
  testTypeChecker(isError, new Error(), 'foo', 'Error')
})

describe('isArray', () => {
  testTypeChecker(isArray, [], 'foo', 'Array')
})

describe('isObject', () => {
  testTypeChecker(isObject, {}, [], 'Object')
})
