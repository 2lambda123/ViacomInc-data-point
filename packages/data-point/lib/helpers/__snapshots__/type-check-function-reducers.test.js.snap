// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isArray It should throw error when not matched 1`] = `"value did not pass type check, type expected: Array but received: string. The value received is: 'foo'"`;

exports[`isBoolean It should throw error when not matched 1`] = `"value did not pass type check, type expected: boolean but received: string. The value received is: 'foo'"`;

exports[`isBoolean It should throw error when not matched 2`] = `"value did not pass type check, type expected: boolean but received: string. The value received is: 'foo'"`;

exports[`isError It should throw error when not matched 1`] = `"value did not pass type check, type expected: function but received: string. The value received is: 'foo'"`;

exports[`isFunction It should throw error when not matched 1`] = `"value did not pass type check, type expected: function but received: string. The value received is: 'foo'"`;

exports[`isNumber It should throw error when not matched 1`] = `"value did not pass type check, type expected: number but received: string. The value received is: 'foo'"`;

exports[`isObject It should throw error when not matched 1`] = `"value did not pass type check, type expected: Object but received: array. The value received is: []"`;

exports[`isString It should throw error when not matched 1`] = `"value did not pass type check, type expected: string but received: number. The value received is: 1"`;
